crates/wasmprinter/src/lib.rs

```diff
@@ -68,8 +70,10 @@ struct Printer<'cfg, 'env> {
 struct CoreState {
     types: Vec<Option<SubType>>,
     funcs: u32,
+    func_to_type: Vec<Option<u32>>,
```

Comment:
> To help prevent some divergence of sources of truth, could funcs be removed here and replaced with a fn funcs(&self) -> u32 { self.func_to_type.len() as u32 } method?

It's not totally trivial, because `CoreState::funcs` is incremented when reading each `Payload::CodeSectionEntry` to keep track of the current `func_idx` (whereas by that point, `func_to_type` has already learned all the function types in the module). But there's probably a simpler way to represent this somehow... let me work on it.

Reply:
Ah ok if it's not easy to do it's ok to leave this as-is. It's not the end of the world to ensure that both are mutated at the same time. I can always try to refactor things as well after this lands.